android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java

```diff
@@ -915,14 +916,18 @@ private String getBadgeText(Cloudlet cloudlet) {
      */
     @Override
     public void onFindCloudlet(final AppClient.FindCloudletReply closestCloudlet) {
+        if (mAppInstanceReplyList != null) {
+            // Clear any existing lines on the map, but re-add existing cloudlets.
+            onGetCloudletList(mAppInstanceReplyList);
+        }
         runOnUiThread(new Runnable() {
             @Override
             public void run() {
                 Cloudlet cloudlet = null;
                 for (int i = 0; i < CloudletListHolder.getSingleton().getCloudletList().size(); i++) {
                     cloudlet = CloudletListHolder.getSingleton().getCloudletList().valueAt(i);
-                    if(cloudlet.getMarker().getPosition().latitude == closestCloudlet.getCloudletLocation().getLatitude() &&
-                            cloudlet.getMarker().getPosition().longitude == closestCloudlet.getCloudletLocation().getLongitude() ) {
+                    Log.i(TAG, "Checking: "+closestCloudlet.getFqdn()+" "+cloudlet.getFqdn());

```

Comment:
Where does closestCloudlet come from?

Reply:
MatchingEnginerHelper > findClodulet

Reply:
Ahh gotcha. What does this code do @brucearmstrong ? Check if the findCloudletReply cloudlet is in the appInstListReply list of cloudlets?

Reply:
At startup, I do a getAppInstList to populate the map with cloudlets. If the user does a findCloudlet, I want to highlight the cloudlet icon (turn it green) and draw a line from the mobile icon to the cloudlet icon. That why I'm searching for a match in the existing list.